File: TeadsSDKDemo/webviewhelper/src/main/java/tv/teads/webviewhelper/LoadJSRunnable.java
<<<<<<< HEAD
=======
/*
 * Copyright (c) Teads 2018.
 */

>>>>>>> origin/update_to_sdk_V4
package tv.teads.webviewhelper;

import android.util.Log;
import android.webkit.WebView;

import java.lang.ref.WeakReference;

/**
 * Created by Benjamin Volland on 20/04/2018.
 */
public class LoadJSRunnable implements Runnable {


    private static final String TAG = WebViewHelper.class.getSimpleName();

    private final WeakReference<WebView> mWeakWebView;
    private final String                 mCommand;

    public LoadJSRunnable(WebView webview, String command) {
        mWeakWebView = new WeakReference<>(webview);
        mCommand = command;
    }

    @Override
    public void run() {
        WebView webView = mWeakWebView.get();
        if (webView != null && webView.getParent() != null) {
            try {
                webView.loadUrl(mCommand);
            } catch (Exception e) {
                Log.w(TAG, "Unable execute the following command: " + mCommand + ", the WebView may have been deallocated. " +
                        "Message: " + e.getMessage());
            }
        }
    }
}

File: TeadsSDKDemo/webviewhelper/src/main/java/tv/teads/webviewhelper/LoadJSRunnable.kt
/*
 * Copyright (c) Teads 2018.
 */

package tv.teads.webviewhelper

import android.util.Log
import android.webkit.WebView

import java.lang.ref.WeakReference

class LoadJSRunnable(webview: WebView, private val command: String) : Runnable {

    private val weakWebView: WeakReference<WebView> = WeakReference(webview)

    override fun run() {
        val webView = weakWebView.get()
        if (webView != null && webView.parent != null) {
            try {
                webView.loadUrl(command)
            } catch (e: Exception) {
                Log.w(TAG, "Unable execute the following command: " + command + ", the WebView may have been deallocated. " +
                        "Message: " + e.message)
            }

        }
    }

    companion object {
        private val TAG = WebViewHelper::class.java.simpleName
    }
}